web/src/elements/controllers/SessionContextController.ts

```diff
@@ -0,0 +1,89 @@
+import type { APIResult } from "#common/api/responses";
+import { EVENT_WS_MESSAGE } from "#common/constants";
+import { isCausedByAbortError, parseAPIResponseError } from "#common/errors/network";
+import { me } from "#common/users";
+
+import { AKConfigMixin, kAKConfig } from "#elements/mixins/config";
+import { SessionContext, SessionMixin } from "#elements/mixins/session";
+import type { ReactiveElementHost } from "#elements/types";
+
+import { SessionUser } from "@goauthentik/api";
+
+import { setUser } from "@sentry/browser";
+
+import { ContextProvider } from "@lit/context";
+import type { ReactiveController } from "lit";
+
+/**
+ * A controller that provides the session information to the element.
+ *
+ * @see {@linkcode SessionMixin}
+ */
+export class SessionContextController implements ReactiveController {
+    #log = console.debug.bind(console, `authentik/controller/session`);
+    #abortController: null | AbortController = null;
+
+    #host: ReactiveElementHost<SessionMixin & AKConfigMixin>;
+    #context: ContextProvider<SessionContext>;
+
+    constructor(
+        host: ReactiveElementHost<SessionMixin & AKConfigMixin>,
+        initialValue?: APIResult<SessionUser>,
+    ) {
+        this.#host = host;
+
+        this.#context = new ContextProvider(this.#host, {
+            context: SessionContext,
+            initialValue: initialValue ?? { loading: true, error: null },
```

Comment:
@BeryJu 

In a future revision we can inject the `/core/users/me/` payload into the Django template like the authentik config and skip a render cycle 😎